# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/renderer/mod.rs
# Commit: 45eeed639240
# Full Hash: 45eeed639240b52db83fbdcf98a2295662d2145c
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2021-10-14 03:02:57
# Description:
#   Bug 1734958 - Don't spin forever trying to get gl errors. r=gfx-reviewers,jnicol
#   
#   I should have known better than to expect that all drivers would implement a queue correctly.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D128385
# ==============================================================================

diff -r 98067437416e -r 45eeed639240 gfx/wr/webrender/src/renderer/mod.rs
--- a/gfx/wr/webrender/src/renderer/mod.rs	Wed Oct 13 15:57:40 2021 +0000
+++ b/gfx/wr/webrender/src/renderer/mod.rs	Wed Oct 13 16:09:17 2021 +0000
@@ -2393,7 +2393,9 @@
     }
 
     fn check_gl_errors(&mut self) {
-        loop {
+        // We shouldn't have to, but make sure we don't iterate more than 50
+        // times, in case a buggy driver decides to never return gl::NO_ERROR.
+        for _ in 0..50 {
             let err = self.device.gl().get_error();
             if err == gl::NO_ERROR {
                 break;
